File: src/ducks/store.js
import { createStore, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import reducer from './reducer';

const persistConfig = { key: 'root', storage };
const persistedReducer = persistReducer(persistConfig, reducer);

const initialState = {
    code: '',
    quiz: [],
    leaderboard: [],
    leaderboards: [],
    answers: [],
    validQuizCodes: [],
    validQuizOptions: { category: [], difficulty: [], type: [] }
}

export default () => {
    const store = createStore(persistedReducer, initialState, applyMiddleware(thunk));
    const persistor = persistStore(store);
    return { store, persistor };
}

File: app/models/Store.scala
package models

case class Store(
                  code: String = "",
                  quiz: Seq[Question] = Seq(),
                  leaderboard: Seq[LeaderboardRow] = Seq(),
                  leaderboards: Seq[Leaderboard] = Seq(),
                  answers: Seq[QuestionAnswer] = Seq(),
                  validQuizCodes: Seq[String] = Seq(),
                  validQuizOptions: ValidQuizOptions = ValidQuizOptions()
                )
